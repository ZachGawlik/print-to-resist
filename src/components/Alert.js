import PropTypes from 'prop-types';
import React from 'react';
import '../styles/Alert.css';

const Alert = ({ children, type }) => (
  <div className={`Alert Alert--${type}`}>
    { children }
  </div>
);

Alert.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string.isRequired,
};

export default Alert;
