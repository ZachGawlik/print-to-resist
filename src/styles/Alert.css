.Alert {
  padding: 20px;
  margin: 20px 5px;
  border: solid 1px black;
  border-radius: 5px;
}

.Alert--error {
  background-color: hsl(0, 50%, 90%);
  border-color: hsl(0, 50%, 80%);
  color: hsl(0, 50%, 30%);
}

.Alert--success {
  background-color: hsl(100, 50%, 90%);
  border-color: hsl(100, 50%, 80%);
  color: hsl(100, 50%, 30%);
}
