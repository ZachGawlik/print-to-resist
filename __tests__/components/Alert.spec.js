import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Alert from '../../src/components/Alert';

describe('Alert', () => {
  it('renders error as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Alert type="error">Bad thing!</Alert>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders success as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Alert type="success">Good thing!</Alert>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
