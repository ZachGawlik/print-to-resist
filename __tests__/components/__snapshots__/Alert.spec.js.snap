// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders error as expected 1`] = `
<div
  className="Alert Alert--error"
>
  Bad thing!
</div>
`;

exports[`Alert renders success as expected 1`] = `
<div
  className="Alert Alert--success"
>
  Good thing!
</div>
`;
